runtime/jvmti/jvmtiClass.c

```diff
@@ -366,13 +366,24 @@ jvmtiGetClassSignature(jvmtiEnv* env,
 					signature[arity] = 'L';
 					memcpy(signature + arity + 1, J9UTF8_DATA(utf), utfLength);
 					signature[allocSize - 1] = ';';
+
+					if (J9ROMCLASS_IS_ANON_OR_HIDDEN(leafType->romClass)) {
+						int i = allocSize - 1;
+						for (; i >= 0; i--) {
+							if (signature[i] == '/') {
```

Comment:
You can use `ANON_CLASSNAME_CHARACTER_SEPARATOR` instead of `/`.

The class name is generated from:

https://github.com/eclipse-openj9/openj9/blob/e8abb75a2c5f91e95a3aca4d60c144875b73f8fb/runtime/bcutil/ROMClassBuilder.cpp#L408-L415

You may want to add an additional check it is `0x<address>` after the last `ANON_CLASSNAME_CHARACTER_SEPARATOR`.

Reply:
updated in a95d7f9